src/Catalog/Registration/RegistrationMaker.cs

```diff
@@ -38,7 +38,16 @@ public static class RegistrationMaker
             IDictionary<RegistrationEntryKey, RegistrationCatalogEntry> promoted = new Dictionary<RegistrationEntryKey, RegistrationCatalogEntry>();
             foreach (var newItem in newItems)
             {
-                promoted.Add(RegistrationCatalogEntry.Promote(newItem.Key, newItem.Value, isExistingItem: false));
+                var promotedEntry = RegistrationCatalogEntry.Promote(newItem.Key, newItem.Value, isExistingItem: false);
+
+                if (!promoted.ContainsKey(promotedEntry.Key))
+                {
+                    promoted.Add(promotedEntry);
+                }
+                else
+                {
+                    promoted[promotedEntry.Key] = promotedEntry.Value;
+                }
```

Comment:
we can replace this code with just:  promoted[promotedEntry.Key] = promotedEntry.Value;